lib/Normalize_std_ast.ml

```diff
@@ -98,6 +98,29 @@ let is_doc = function
 let is_erasable_jane_syntax attr =
   String.is_prefix ~prefix:"jane.erasable." attr.attr_name.txt
 
+(* Immediate layout annotations should be treated the same as their attribute
+   counterparts *)
+let normalize_immediate_annot_and_attrs attr =
+  match (attr.attr_name.txt, attr.attr_payload) with
+  (* CR layouts: change to something like: {[ | (
+     "jane.erasable.layouts.annot" , PStr [ { pstr_desc= Pstr_eval
+     ({pexp_desc= Pexp_ident {txt= Lident "immediate"; _}; _}, _) ; _ } ] )
+     -> attr ]} after the parsing logic catches up to what's in
+     flambda-backend. *)
+  (* We also have to normalize "ocaml.immediate" into "immediate"
+     for this to work. Since if we rewrite [@@ocaml.immediate] into
+     an annotation and treat that as [@@immediate]. That's an attribute
+     change we need to accept. *)
+  | "jane.erasable.layouts.immediate", _ | "ocaml.immediate", _ ->
```

Comment:
This has the potential to drop payloads. There shouldn't be payloads of course... but I don't think we should drop them.

Reply:
That's a good point! I changed it to only do the rewrite when there's no attribute payload and added a test case to show that.